src/skvhttpclient/test/ClientIntegrationTest.cpp

```diff
@@ -350,6 +363,101 @@ void testQuery1() {
+        auto key = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"));
+
+        auto&& [writeStatus] = txn.write(record1).get();
+        K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), true);
+        auto&& [readStatus, readRecord] = txn.read(key).get();
+        K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+        verifyEqual(readRecord, record1); // Not working, expected 35 getting 33
+        {
+            K2LOG_I(k2::log::httpclient, "Write with existence precondition not exists, shold fail with 412 code: record exists");
+            auto record1_1 = buildRecord(collectionName, schemaPtr, std::string("A1"), std::string("B1"), 36, std::string("Test1_Update2"));
+            auto&& [writeStatus] = txn.write(record1_1, false, dto::ExistencePrecondition::NotExists).get();
+            K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(),false);
+            K2EXPECT(k2::log::httpclient, writeStatus.code, 412);
+            auto&& [readStatus, readRecord1] = txn.read(key).get();
+            K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+            verifyEqual(readRecord1, record1);
+        }
+    }
+
+    auto key1 = buildRecord(collectionName, schemaPtr, std::string("A2"), std::string("B2"));
+    {
+        K2LOG_I(k2::log::httpclient, "Write record with no value");
+        // Write a record with no value for Data field, should fail
+        auto record1 = buildRecord(collectionName, schemaPtr, std::string("A2"), std::string("B2"), 32);
+        auto&& [writeStatus] = txn.write(record1).get();
+        K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), false);
+    }
+    {
+        // Write with Data = null
+        auto record1 = buildRecord(collectionName, schemaPtr, std::string("A2"), std::string("B2"), 32, std::optional<std::string>());
+        auto&& [writeStatus] = txn.write(record1).get();
+        K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), true);
+        auto&& [readStatus, readRecord] = txn.read(key1).get();
+        K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+        verifyEqual(readRecord, record1);
+    }
+    {
+        // Update data field from null to new value
+        auto record1 = buildRecord(collectionName, schemaPtr, std::string("A2"), std::string("B2"), 34, std::string("Test2_Update"));
+        std::vector<uint32_t> fields = {3}; // Update data field only
+        auto&& [writeStatus] = txn.partialUpdate(record1, fields).get();
+        K2EXPECT(k2::log::httpclient, writeStatus.is2xxOK(), true);
+        auto&& [readStatus, readRecord] = txn.read(key1).get();
+        K2EXPECT(k2::log::httpclient, readStatus.is2xxOK(), true);
+        auto compareRecord = buildRecord(collectionName, schemaPtr, std::string("A2"), std::string("B2"), 32, std::string("Test2_Update"));
+        verifyEqual(readRecord, compareRecord);
+    }
+
+    {
+        K2LOG_I(k2::log::httpclient, "Partial update with precondition test");
```

Comment:
We can remove this test since the user shouldn't specify a dto::ExistencePrecondition for a partial update

Reply:
Fixed